internal/pkg/instrumentation/probe/probe.go

```diff
@@ -17,39 +17,53 @@ import (
 
 	"github.com/Masterminds/semver/v3"
 	"github.com/cilium/ebpf"
-	"github.com/cilium/ebpf/link"
 	"github.com/cilium/ebpf/perf"
 
 	"go.opentelemetry.io/collector/pdata/pcommon"
 	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.opentelemetry.io/otel/trace"
 
 	"go.opentelemetry.io/auto/internal/pkg/inject"
-	"go.opentelemetry.io/auto/internal/pkg/instrumentation/bpffs"
 	"go.opentelemetry.io/auto/internal/pkg/instrumentation/probe/sampling"
-	"go.opentelemetry.io/auto/internal/pkg/instrumentation/utils"
 	"go.opentelemetry.io/auto/internal/pkg/process"
 	"go.opentelemetry.io/auto/internal/pkg/structfield"
 	"go.opentelemetry.io/auto/pipeline"
 )
 
+// ID is a unique identifier for a probe.
+type ID struct {
+	// SpanKind is the span kind handled by the probe.
+	SpanKind trace.SpanKind
+	// InstrumentedPkg is the package path of the instrumented code.
+	InstrumentedPkg string
+}
+
+func (id ID) String() string {
+	return fmt.Sprintf("%s/%s", id.InstrumentedPkg, id.SpanKind)
+}
+
 // Probe is the instrument used by instrumentation for a Go package to measure
 // and report on the state of that packages operation.
 type Probe interface {
-	// Manifest returns the Probe's instrumentation Manifest. This includes all
-	// the information about the package the Probe instruments.
-	Manifest() Manifest
```

Comment:
Replacing this with individual mentions does not seem advantageous. 

Methods cannot be added to an interface in a backwards compatible manner. Fields and methods can be added to a strict in a backwards compatible manner.

There is also no way for default behavior for an interface other than compile time failures. A struct can be designed to apply sane defaults are expressive errors. 

I feel we should continue to hold static definitions about the probe in a manifest type of some sort instead of this flattening. Methods for the probe should be added for actions the probe makes and other components will can to initiate those actions.

Reply:
@MrAlias I re-added the `Manifest()` method to the Probe interface after our discussion on the sig call. I tried taking this a little step further and now the Manifest just stores the raw `Consts` and `Uprobes`, with methods for getting the `StructFields` and `Symbols` specifically from that. We had talked before about trying to align these terms, and they do seem to have separate usages, so I'm curious what you think

Right now we precompute and sort these fields, which I could see being an advantage to not have to sort them every time they are called. So, maybe we store consts, uprobes, structfields, *and* symbols in the Manifest. The tradeoff there is just the redundancy. So either way could work imo